src/metadata/utils.ts

```diff
@@ -57,3 +60,10 @@ export function ensureReflectMetadataExists() {
     throw new ReflectMetadataMissingError();
   }
 }
+
+export function flattenExtensions(
```

Comment:
I am not very happy with this, having an "util" for just a specific field does not make a lot of sense to me but I also didn't want to repeat the reducer function twice.

I can propose two alternatives:
* make a more generic util which will flatten by key passed as argument (it seemed a little bit premature to me since there are no other cases in the codebase yet, but I could do it)
* simply use an inline function duplicated twice and remove this util

What do you think?

Reply:
Yes, you should just inline the spread operator usage, it makes no sense to wrap single line in dedicated 5 lines function

Reply:
Thanks for the feedback, I have changed it in this commit: 6eac60ee487040f4072de05259e28539afe9bdf5

Please let me know if you'd rather have me wait until your review is over before I start addressing your comments 🙂